### ISSUE TITLE
Basic repository setup

### ISSUE THREAD

**Issue by rmandelb** [LSSTDESC/DC2-analysis#1]
This brand new shiny repository needs some help:

- [x] LICENSE
- [x] More descriptive README explaining what kinds of things go here vs. DC2-production vs. project-specific repositories
- [x] Statements in the README about the state of the code
- [x] A reasonable set of labels on the Issues page
- [x] Protect master branch

Moved elsewhere:
- Transfer of existing analysis codes from DC2-production - moved to #2 
- Add instruction for how to upload a notebook to this repo - moved to #3 
- A sensible directory structure  - moved to #3.

**Comment by yymao** [comment]
- [ ] Add instruction for how to upload a notebook to this repo.

A related question, do we want to protect master branch or do we want to make it easier for people to upload their notebook (i.e., directly to master)?

**Comment by rmandelb** [comment]
@yymao - I've updated the comment in the first comment in this thread to reflect which tasks have been moved to their own issues.

Also, I made a start at a set of labels.  I thought that if we're mostly including analysis scripts here, then we can have labels that indicate (a) the type of data (e.g., "data:extragalactic"), (b) the access method (e.g. "access:butler"), and (c) the type of science it's relevant for (e.g., "analysis:PZ").  So if there's a script that's about reading in coadd data using the butler and plotting some WL quantities, it would get data:coadd, access:butler, analysis:WL.  If something is just about doing basic tests of accessing extragalactic catalogs using catalogs, it would get data:extragalactic, access:catalog (no science-related label).

The set of labels according to this scheme isn't yet complete, I just put a few examples of each type of data / access / analysis.  Thoughts on this are welcome.

**Comment by yymao** [comment]
Thanks, @rmandelb! That sounds good. Two minor points about naming: 
- Does `access:catalog` mean the same thing as `access:gcr`? 
- We are in fact renaming the "coadd catalog" to the "object catalog", so maybe we should label it `data:object`? (cc @drphilmarshall @wmwv, cf https://github.com/LSSTDESC/gcr-catalogs/pull/165)

**Comment by rmandelb** [comment]
It depends to a certain extent whether the labels are for the experts, who want to use the language they are already using to talk about the scripts, or for others who don't have all the jargon down.  GCR is the tool we'll use for accessing catalogs in these scripts, but people who are new to DC2 and try to find tutorials that are relevant to them by browsing the issue page will probably find `access:catalog` more intuitively understandable (we're accessing the data by reading catalogs) than `access:gcr` (we're accessing the data by... some acronym?).  There is of course a limit to how much jargon can reasonably be removed from the labels, so if there's a significant preference to use actual tool names consistently in all cases I'm fine with it.  

We should be consistent across the repos, so if we're going to self-consistently rename the coadd catalog to something else, we should do it here too.

**Comment by yymao** [comment]
I was thinking that because `gcr` appears in many of the tutorial names, it might be easier for people to identify? Anyway, if `access:catalog` does mean using GCR but we just want to avoid acronym, then I would suggest that we use `access:reader`.  

Yes, we have started the process of renaming "coadd" --> "object". Unfortunately we failed to do this *before* the tutorial, so there will be a period of time these two terms appear together. @wmwv can correct me but I think the plan is to consistently use "object" when releasing 1.2 data to the Collaboration.
